src/vm/eepolicy.cpp

```diff
@@ -1200,6 +1200,20 @@ inline void DoLogForFailFastException(LPCWSTR pszMessage, PEXCEPTION_POINTERS pE
         {
             PrintToStdErrA("\n");
             LogCallstackForLogWorker();
+            EXCEPTIONREF refException = (EXCEPTIONREF)(pThread->LastThrownObject());
```

Comment:
The LastThrowObject is not necessarily associated with the reason why failfast is called. The exception should be passed in as argument as argument when it is meant to be logged.